packages/signer/src/Signer.js

```diff
@@ -0,0 +1,273 @@
+        this.hashAlgorithm = await this.getHashAlgorithm();
+        // Get a crypto extension
+        this.crypto = this.getCrypto();
+        // Get the signing (and verification) certificates
+        const certificates = await this.obtainCertificates();
+        const signCert = certificates[0];
+        // Obtain the private key used for signing
+        const key = await this.obtainKey();
+
+        // Creation of the CMS Signed Data structure (see [PDF - 12.8.3.3] and [PAdES - 4.1])
+        // Setup signer info (see [CMS - 5.3])
+        const signerInfo = new pkijs.SignerInfo({
+            version: 1,
+            sid: new pkijs.IssuerAndSerialNumber({
+                issuer: signCert.issuer,
+                serialNumber: signCert.serialNumber,
+            }),
+            signedAttrs: new pkijs.SignedAndUnsignedAttributes({
+                type: 0,
+                attributes: await this.obtainSignedAttributes(signingTime, pdfBuffer, signCert),
+            }),
+        });
+
+        // Initialize CMS Signed Data structure (see [CMS - 5.1]) and sign it
+        const cmsSignedData = new pkijs.SignedData({
+            version: 1,
+            encapContentInfo: new pkijs.EncapsulatedContentInfo({
+                eContentType: oids.data, // No actual econtent here, as we sign in detached mode
+            }),
+            signerInfos: [signerInfo],
+            certificates,
+        });
+
+        // Sign in detached mode. That's what Adobe.PPKLite expects for subfilters
+        // adbe.pkcs7.detached and ETSI.CAdES.detached.
+        await cmsSignedData.sign(key, 0, this.hashAlgorithm, undefined, this.crypto);
+
+        // Append the unsigned attributes, if there are any
+        const unsignedAttrs = await this.obtainUnsignedAttributes(signerInfo.signature.getValue());
+        // if (unsignedAttrs.length > 0) {
+        //     signerInfo.unsignedAttrs = new pkijs.SignedAndUnsignedAttributes({
+        //         type: 1,
+        //         attributes: unsignedAttrs,
+        //     });
+        // }
+
+        // Create final result
+        const cmsContentWrap = new pkijs.ContentInfo({
+            contentType: oids.signedData,
+            content: cmsSignedData.toSchema(true),
+        });
+
+        return Buffer.from(cmsContentWrap.toSchema().toBER(false));
+    }
+
+    /**
+     * Verify whether the signature generated by the sign function is correct.
+     * @param {Buffer} cmsSignedBuffer
+     * @param {Buffer} pdfBuffer
+     * @returns {boolean}
+     */
+    async verify(cmsSignedBuffer, pdfBuffer) {
+        // Based on cmsSignedComplexExample from PKI.js
+        const cmsContentSimpl = pkijs.ContentInfo.fromBER(cmsSignedBuffer);
+        const cmsSignedSimpl = new pkijs.SignedData({schema: cmsContentSimpl.content});
+
+        return cmsSignedSimpl.verify({
+            signer: 0,
+            trustedCerts: [],
+            data: pdfBuffer,
+        }, this.getCrypto());
+    }
```

Comment:
Is there any real-world use for this function in our library or is it just a testing helper? I'd suggest there's no need to provide a `verify` method as part of PDF signing, unless we are also offering PDF verification too

Reply:
No it's for testing purposes only.